### PR TITLE
HLRC XPack Protocol clean up: Migration; Graph; Watcher

### PR DESCRIPTION
**Pull request by vladimirdolzhenko**
Move parts related to XPack for packages Migration; Graph; Watcher from protocol to client

Drop protocol module

Relates #34451

### CODE REVIEW

File: x-pack/protocol/src/test/java/org/elasticsearch/protocol/xpack/migration/IndexUpgradeInfoResponseTests.java [hunk 1]
```diff
@@ -1,68 +0,0 @@
...
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.test.AbstractStreamableXContentTestCase;
-
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-
-public class IndexUpgradeInfoResponseTests extends AbstractStreamableXContentTestCase<IndexUpgradeInfoResponse> {
-    @Override
-    protected IndexUpgradeInfoResponse doParseInstance(XContentParser parser) {
```
[the remark this answers is not in the document]
Comment (vladimirdolzhenko) [reply x-pack/protocol/src/test/java/org/elasticsearch/protocol/xpack/migration/IndexUpgradeInfoResponseTests.java]: @nik9000 we can keep it `ToXContentObject` in this PR and keep this test as-is, in #34547 I add cross xpack-hlrc compatibility tests that makes `ToXContentObject` for hlrc objects a bit of redundant

[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/protocol/src/test/java/org/elasticsearch/protocol/xpack/migration/IndexUpgradeInfoResponseTests.java]: :+1:

File: client/rest-high-level/src/test/java/org/elasticsearch/client/migration/IndexUpgradeInfoResponseTests.java [hunk 2]
```diff
@@ -17,7 +17,10 @@
...
-/**
- * Request and Response objects for miscellaneous X-Pack APIs.
- */
-package org.elasticsearch.protocol.xpack;
+package org.elasticsearch.client.migration;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class IndexUpgradeInfoResponseTests extends ESTestCase {
+    // TODO: add to cross XPack-HLRC serialization test
```
Comment (nik9000) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/migration/IndexUpgradeInfoResponseTests.java]: :+1: I see you already have that in flight in another PR so this is fine by me!

**Review by nik9000 — APPROVED** [review]
In combination with the other in flight PRs around testing, this looks good to me.

File: client/rest-high-level/build.gradle [hunk 3]
```diff
@@ -51,7 +51,6 @@ dependencies {
   compile "org.elasticsearch.plugin:aggs-matrix-stats-client:${version}"
   compile "org.elasticsearch.plugin:rank-eval-client:${version}"
   compile "org.elasticsearch.plugin:lang-mustache-client:${version}"
-  bundle project(':x-pack:protocol')
```
Comment (hub-cap) [inline client/rest-high-level/build.gradle]: :tada:

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeleteWatchRequest.java [hunk 4]
```diff
@@ -58,28 +55,14 @@ public void setId(String id) {
     }
 
     @Override
-    public ActionRequestValidationException validate() {
-        ActionRequestValidationException validationException = null;
-        if (id == null){
-            validationException = ValidateActions.addValidationError("watch id is missing", validationException);
+    public Optional<ValidationException> validate() {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeleteWatchRequest.java]: lets put this in the constructor instead if possible.

https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeactivateWatchRequest.java#L29-L37

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/PutWatchRequest.java [hunk 5]
```diff
@@ -136,20 +109,20 @@ public void setVersion(long version) {
     }
 
     @Override
-    public ActionRequestValidationException validate() {
-        ActionRequestValidationException validationException = null;
+    public Optional<ValidationException> validate() {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/PutWatchRequest.java]: can we move this to the constructor a-la https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeactivateWatchRequest.java#L29-L37

**Review by hub-cap — APPROVED** [review]
Great work. @rjernst will be jealous that you got to remove so much code instead of him ;) One small nit on 2 watcher classes but i dont need to see this again.

**Comment by nik9000** [comment]
:heart:

**Pull request merged**